core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala

```diff
@@ -1415,6 +1415,79 @@ class BlockManagerSuite extends SparkFunSuite with Matchers with BeforeAndAfterE
       super.fetchBlockSync(host, port, execId, blockId)
     }
   }
+
+  def testGetOrElseUpdateForLargeBlock(storageLevel: StorageLevel) {
```

Comment:
@cloud-fan  I know, 
it even gets worse when using the `===` operator.

I'm currently exploring the second direction pointed by @vanzin , introducing a test-only configuration key to configure the max page size